### ISSUE TITLE
[Feature Request] Disable then enable

### ISSUE THREAD

**Issue by lonix1** [koalaman/shellcheck#2512]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I couldn't find anything similar other than #2196 which is a "global" toggle, and not what I'm referring to here.

### Problem
In many static analysers you can do this
````c
// disable 1234
var something = that.Causes(theError);        // error is not reported
and.this.AlsoCausesTheError();                // error is not reported
// enable 1234
foo.causeErrorAgain();                        // error is reported
````

So we can disable a rule for one or more lines, then enable it again for the rest of the file.

### Feature Request
Please consider something like this for shellcheck:
````sh
# shellcheck disable=SC1083
...
# shellcheck enable=SC1083
````

**Comment by leagris** [comment]
Alternatively put the SC disable commands into a commands block that shares the same shellcheck disabled entries:

```sh
# shellcheck disable=SCxxx
{
  var something = that.Causes(theError);  // error is not reported
  and.this.AlsoCausesTheError();                // error is not reported
}
foo.causeErrorAgain();                               // error is reported
``

**Comment by lonix1** [comment]
That is a nice trick @leagris, thanks.

Sadly it doesn't work when using templating, as discussed [here](https://github.com/koalaman/shellcheck/issues/1306), but it's a nice workaround in other cases.

**Comment by lpar** [comment]
This would be useful for my use case too, which is that I have a shell script with a man page embedded in it.

**Comment by woodward2** [comment]
Came across an instance of this yesterday..

I was thinking something like:
`# shellcheck disable=SC1234:5`
meaning disable the check for the next 5 lines.

**Comment by prahal** [comment]
Note that disable does not disable a rule per https://github.com/koalaman/shellcheck/wiki/Directive#disable, it only silences the error, which means parsing stops at the same point in the file. Therefore, enabling the error again after the point where it was silenced is not possible, because parsing has stopped.
